archai/nlp/objectives/parameters.py

```diff
@@ -39,24 +30,21 @@ def evaluate(self, model: ArchaiModel, dataset: DatasetProvider, budget: Optiona
         return total_params
 
 
-class NonEmbeddingParamsProxy(Objective):
-    """Implement an objective that calculates the number of non-embedding parameters
-    of a PyTorch model.
-
-    """
-
-    higher_is_better: bool = True
-
+class NonEmbeddingParamsProxy(SyncEvaluator):
     def __init__(self, exclude_cls: Optional[List[nn.Module]] = None, trainable_only: Optional[bool] = True) -> None:
-        """Initialize the `NonEmbeddingParamsProxy` instance.
+        """Total number of non-embedding parameters.
+            Used as a proxy for the perplexity of decoder-only transformer LMs.
 
-        Args:
-            exclude_cls: A list of `nn.Module` classes to be ignored
-                during parameter counting.
-            trainable_only: A flag indicating whether only trainable parameters
-                should be counted.
+            From: "LiteTransformerSearch: Training-free Neural Architecture Search for
+                Efficient Language Models", Javaheripi et. al, 2022
 
-        """
+        Args:
+            exclude_cls (Optional[List[nn.Module]], optional): List of PyTorch module classes
```

Comment:
Do you think that we should use annotated typing on the docstrings?

I have seen on this guide that it may not be necessary since we are already annotating the headers: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

```
def function_with_pep484_type_annotations(param1: int, param2: str) -> bool:
    """Example function with PEP 484 type annotations.

    Args:
        param1: The first parameter.
        param2: The second parameter.

    Returns:
        The return value. True for success, False otherwise.

    """
```

Reply:
That's a good question. I like having them there too, but they can be tricky to deal with when the type expressions are lengthy...

Reply:
That's true!

I will leave it up to you, please feel free to choose the style that we will be using. Just please let me know, so I can make the proper changes on other files 😄